source/dub/internal/colorize/winterm.d

```diff
@@ -0,0 +1,191 @@
+/**
+ * Authors: ponce
+ * Date: July 28, 2014
+ * License: Licensed under the MIT license. See LICENSE for more information
+ * Version: 1.0.2
+ */
+module dub.internal.colorize.winterm;
```

Comment:
Compared to what [DMD does](https://github.com/dlang/dmd/blob/master/src/dmd/console.d), this looks overly complicated.
If all this logic is really required for Windows, it would be great to have some explanation for it (and future readers).

Reply:
I agree the approach of d-colorize is indeed more complicated, also on Windows the performance will be a bit lower, since this module is doing this sort of emulation we see here. Maybe at this point it would make more sense to just bring in DMD's `console` module and ditch this mess? :smile:

Reply:
Dmd's Code is well-tested and only a quarter of d-colorize, so this means a lot less maintenance cost. So I would be in favor if you don't mind throwing away a bit of your work.

Reply:
No problem, I'll definitely switch to DMD's `console` module, it's a better decision indeed. Also, that shouldn't be a lot of work to do. Sorry for not doing that in the first place, I'm a bit new to D and its community and I didn't even think about checking DMD's code before, the first thing I did was search for a dub package that did the job, that's how I ended up with d-colorize. I'll make the changes in the next days/weeks, I have some exams pretty soon unfortunately.

Reply:
`d-colorize` does it this way in order to have color codes inside the strings...

> now I see why they went with that approach

Indeed, that's why I went with this approach, after too much time lost with colorizing console output like DMD does.